hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/AbstractWriteHelper.java

```diff
@@ -79,7 +79,7 @@ public I combineOnCondition(
    *
    * @param records     hoodieRecords to deduplicate
    * @param parallelism parallelism or partitions to be used while reducing/deduplicating
-   * @return RDD of HoodieRecord already be deduplicated
+   * @return Collection of HoodieRecord already be deduplicated
    */
   public I deduplicateRecords(
       I records, HoodieTable<T, I, K, O> table, int parallelism) {
```

Comment:
Hi, @yanghua Thanks for raising this pr.
LGTM overall. and I noticed that there are still two lines contains `RDD` in `hudi-client-common` module. maybe we can improve them together.
hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java:76
hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java:121

Reply:
Agree to remove the legacy comment for `HoodieIndexUtils `. IMO, we can remain the comment in `HoodieWriteConfig` as a description of that config option.

Reply:
> Agree to remove the legacy comment for `HoodieIndexUtils `. IMO, we can remain the comment in `HoodieWriteConfig` as a description of that config option.

Yes, thanks for addressing my concern.
LGTM now